### PR TITLE
Create workload cluster from management cluster for baremetal

### PR DESCRIPTION
**Pull request by panktishah26**
*Issue #, if available:*

*Description of changes:*
Currently a user can only create a management cluster for the provider Tinkerbell. This PR will add creating workload cluster from management cluster functionality for Tinkerbell. 

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/executables/kubectl.go [hunk 1]
```diff
@@ -1205,6 +1215,26 @@ func (k *Kubectl) SearchVsphereMachineConfig(ctx context.Context, name string, k
 	return response.Items, nil
 }
 
+// SearchTinkerbellMachineConfig returns the list of TinkerbellMachineConfig in the cluster.
+func (k *Kubectl) SearchTinkerbellMachineConfig(ctx context.Context, name string, kubeconfigFile string, namespace string) ([]*v1alpha1.TinkerbellMachineConfig, error) {
```
Comment (chrisdoherty4) [inline pkg/executables/kubectl.go:1219]: Are we searching? This looks like standard retrieval, i.e. `Get*` or `Retrieve*`.

File: pkg/providers/tinkerbell/create.go [hunk 2]
```diff
@@ -155,13 +182,63 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
...
 	}
 
+	if p.clusterConfig.IsManaged() {
+		return p.AddHardwareInCluster(ctx, clusterSpec.ManagementCluster)
+	}
+
+	return nil
+}
+
+func (p *Provider) getHardwaresFromManagementCluster(ctx context.Context, cluster *types.Cluster) error {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/create.go:201]:
```suggestion
func (p *Provider) getHardwareFromManagementCluster(ctx context.Context, cluster *types.Cluster) error {
```

File: pkg/providers/tinkerbell/tinkerbell_test.go [hunk 3]
```diff
@@ -46,6 +49,16 @@ func givenMachineConfigs(t *testing.T, fileName string) map[string]*v1alpha1.Tin
 	return machineConfigs
 }
 
+func thenErrorExpected(t *testing.T, expected string, err error) {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/tinkerbell_test.go:52]:
```suggestion
func assertError(t *testing.T, expected string, err error) {
```

File: pkg/providers/tinkerbell/create.go [hunk 4]
```diff
@@ -62,6 +62,11 @@ func (p *Provider) PreCAPIInstallOnBootstrap(ctx context.Context, cluster *types
 }
 
 func (p *Provider) PostBootstrapSetup(ctx context.Context, clusterConfig *v1alpha1.Cluster, cluster *types.Cluster) error {
+	return p.AddHardwareInCluster(ctx, cluster)
+}
+
+// AddHardwareInCluster adds all the hardwares provided in the hardware.csv file to the cluster.
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/create.go:68]: I think we should avoid mentioning the hardware CSV in the comment given this code should be oblivious to the use of a hardware CSV.

File: pkg/providers/tinkerbell/create.go [hunk 5]
```diff
@@ -62,6 +62,11 @@ func (p *Provider) PreCAPIInstallOnBootstrap(ctx context.Context, cluster *types
...
+func (p *Provider) AddHardwareInCluster(ctx context.Context, cluster *types.Cluster) error {
```
Comment (chrisdoherty4) [inline pkg/providers/tinkerbell/create.go:69]:
```suggestion
func (p *Provider) ApplyHardwareToCluster(ctx context.Context, cluster *types.Cluster) error {
```

Comment (chrisdoherty4) [reply pkg/providers/tinkerbell/create.go:69]: Also, does this need to be exported?

File: pkg/executables/kubectl.go (hunk 1 above)
Comment (panktishah26) [reply pkg/executables/kubectl.go:1219]: I wanted to keep the name in sync with other providers. https://github.com/aws/eks-anywhere/blob/main/pkg/executables/kubectl.go#L1189
I am fine either ways though.

File: pkg/executables/kubectl_test.go [hunk 6]
```diff
@@ -2836,3 +2836,77 @@ func TestKubectlDeleteEksaNutanixMachineConfigError(t *testing.T) {
...
+		t.Errorf("Kubectl.SearchTinkerbellMachineConfig() error = %v, want nil", err)
+	}
+	if len(mc) > 0 {
+		t.Errorf("expected 0 machine configs, got %d", len(mc))
+	}
+}
+
+func TestKubectlSearchTinkerbellMachineConfigInvalid(t *testing.T) {
+	var kubeconfig, namespace, name string
+	machineconfigJSON := test.ReadFile(t, "testdata/kubectl_tinkerbellmachineconfig_invalid.json")
```
Comment (chrisdoherty4) [inline pkg/executables/kubectl_test.go:2862]: We go to the effort of loading a test file but we don't seem to do any checks to ensure we got the content we expected.

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**